internal/logger/io_sink.go

```diff
@@ -27,71 +29,26 @@ func NewIOSink(out io.Writer) *IOSink {
 	}
 }
 
-func logCommandMessageStarted(log *log.Logger, kvMap map[string]interface{}) {
-	format := "Command %q started on database %q using a connection with " +
-		"server-generated ID %d to %s:%d. The requestID is %d and " +
-		"the operation ID is %d. Command: %s"
-
-	log.Printf(format,
-		kvMap["commandName"],
-		kvMap["databaseName"],
-		kvMap["serverConnectionId"],
-		kvMap["serverHost"],
-		kvMap["serverPort"],
-		kvMap["requestId"],
-		kvMap["operationId"],
-		kvMap["command"])
-
-}
-
-func logCommandMessageSucceeded(log *log.Logger, kvMap map[string]interface{}) {
-	format := "Command %q succeeded in %d ms using server-generated ID " +
-		"%d to %s:%d. The requestID is %d and the operation ID is " +
-		"%d. Command reply: %s"
-
-	log.Printf(format,
-		kvMap["commandName"],
-		kvMap["duration"],
-		kvMap["serverConnectionId"],
-		kvMap["serverHost"],
-		kvMap["serverPort"],
-		kvMap["requestId"],
-		kvMap["operationId"],
-		kvMap["reply"])
-}
-
-func logCommandMessageFailed(log *log.Logger, kvMap map[string]interface{}) {
-	format := "Command %q failed in %d ms using a connection with " +
-		"server-generated ID %d to %s:%d. The requestID is %d and " +
-		"the operation ID is %d. Error: %s"
-
-	log.Printf(format,
-		kvMap["commandName"],
-		kvMap["duration"],
-		kvMap["serverConnectionID"],
-		kvMap["serverHost"],
-		kvMap["serverPort"],
-		kvMap["requestId"],
-		kvMap["operationId"],
-		kvMap["failure"])
-}
-
+// Info will write the provided message and key-value pairs to the io.Writer
+// as extended JSON.
 func (osSink *IOSink) Info(_ int, msg string, keysAndValues ...interface{}) {
 	kvMap := make(map[string]interface{})
+	kvMap[KeyMessage] = msg
+
 	for i := 0; i < len(keysAndValues); i += 2 {
 		kvMap[keysAndValues[i].(string)] = keysAndValues[i+1]
 	}
 
-	switch msg {
-	case CommandStarted:
-		logCommandMessageStarted(osSink.log, kvMap)
-	case CommandSucceeded:
-		logCommandMessageSucceeded(osSink.log, kvMap)
-	case CommandFailed:
-		logCommandMessageFailed(osSink.log, kvMap)
+	kvBytes, err := bson.MarshalExtJSON(kvMap, false, false)
```

Comment:
Based on the latest discussion about the logging spec, it sounds like the expected serialization format is JSON or Extended JSON when writing to a byte stream (e.g. STDOUT or a file). JSON and Extended JSON both encode sufficient information for the types included in log messages (excluding the command/reply document, which is encoded separately). We should prefer using the `"encoding/json"` library here because it is likely to be better supported.

Also, if we're switching to structured logs, we should not use `log.Println` because it will add unstructured data to the log line. Instead, we should add a timestamp to the message and write the JSON messages directly to the output stream.

E.g.
```go
type IOSink struct {
	enc *json.Encoder
}

func NewIOSink(out io.Writer) *IOSink {
	return &IOSink{
		// TODO: Should we wrap the writer in a bufio.Writer? Or delegate that to the caller?
		enc: json.NewEncoder(out),
	}
}

func (osSink *IOSink) Info(_ int, msg string, keysAndValues ...interface{}) {
	// ...

	// I'm assuming that we want to use a nanosecond-resolution Unix timestamp,
	// but the logging spec currently specify.
	kvMap[KeyTimestamp] = time.Now().UnixNano()

	// Logging is best-effort and we can't do anything with this error, so ignore it.
	_ = osSink.enc.Encode(kvMap)
}
```